x-pack/platform/plugins/shared/security/server/uiam/uiam_service.ts

```diff
@@ -157,6 +225,78 @@ export class UiamService implements UiamServicePublic {
     }
   }
 
+  /**
+   * See {@link UiamServicePublic.grantApiKey}.
+   */
+  async grantApiKey(authorization: HTTPAuthorizationHeader, params: GrantUiamAPIKeyParams) {
+    try {
+      this.#logger.debug(
+        `Attempting to grant API key using authorization scheme: ${authorization.scheme}`
+      );
+
+      const body: GrantUiamApiKeyRequestBody = {
+        description: params.name,
+        internal: true,
+        ...(params.expiration ? { expiration: params.expiration } : {}),
+        role_assignments: {
+          limit: {
+            access: ['application'],
+            resource: ['project'],
+          },
```

Comment:
__nit:__ let's clarify in a code comment what this limit does exactly to help future readers.